### PR TITLE
Add support for Rails 7.2

### PR DESCRIPTION
**Pull request by mcelicalderon**
Adds support for Rails 7.2 also tests with Ruby 3.1 as it's required for Rails 7.2

Fixed remaining deprecation warnings in the latest versions of the test matrix

Also fixed some dependencies with appraisals

### CODE REVIEW

**Comment by mcelicalderon** [comment]
@00dav00 can you please review and merge if everything looks good?

File: spec/requests/graphql_controller_spec.rb [hunk 1]
```diff
@@ -33,17 +33,6 @@
     end
   end
 
-  context 'when variables are not a string or hash' do
```
Comment (00dav00) [inline spec/requests/graphql_controller_spec.rb:36]: Do you remember why was this added in the first place?
I'm checking the commit but it doesn't ring any bells :sweat_smile:

**Review by 00dav00 — APPROVED** [review]

File: spec/requests/graphql_controller_spec.rb (hunk 1 above)
Comment (mcelicalderon) [reply spec/requests/graphql_controller_spec.rb:36]: Yes, I added that just to increase coverage, but not sure how the Argument error was being raised in a request spec. Anyway, we need to fix the integration with coveralls anyway as I think it might not be setup properly with their latest changes. Also, the GQL gem probably has a different method for variable parsing now that might not raise an argument error. Thanks for the review
